apps/staking/app/stake/[address]/NewStake.tsx

```diff
@@ -128,27 +127,34 @@ export function NewStake({ contract }: { contract: ContributionContract }) {
   const onSubmit = (data: StakeFormSchema) => {
     setIsSubmitting(true);
 
-    /** We want to re-check the banned list validations*/
-    if (data.rewardsAddress) {
-      if (!isAddress(data.rewardsAddress)) {
+    let rewardsAddress = data.rewardsAddress;
+    if (rewardsAddress) {
+      if (bannedRewardsAddresses.some(({ address }) => areHexesEqual(address, rewardsAddress))) {
         form.setError('root', {
           type: 'manual',
-          message: dictionaryRewardsAddress('invalidAddress'),
+          message: dictionaryRewardsAddress('bannedVestingContract'),
         });
         return;
       }
-
-      if (
-        bannedRewardsAddresses.some(({ address }) => areHexesEqual(address, data.rewardsAddress))
-      ) {
+      // If there is a vesting contract the rewards address is required
+    } else if (vestingContract) {
+      form.setError('root', {
+        type: 'manual',
+        message: dictionaryRewardsAddress('invalidAddress'),
+      });
+      return;
+    } else {
+      if (!connectedAddress) {
         form.setError('root', {
           type: 'manual',
-          message: dictionaryRewardsAddress('bannedVestingContract'),
+          message: dictionaryRewardsAddress('invalidAddress'),
         });
         return;
       }
-      // If there is a vesting contract the rewards address is required
-    } else if (vestingContract) {
+      rewardsAddress = connectedAddress;
+    }
+
+    if (!isAddress(rewardsAddress)) {
```

Comment:
Just wondering but should we not validate the address before we start line 131. Same for `apps/staking/app/stake/[address]/ManageStakeContribution.tsx`